internal/handler/executor.go

```diff
@@ -134,13 +138,49 @@ func (e *Executor) execHookCommand(w io.Writer) error {
 		// log only envs set by webhook, not global env; otherwise it's leaking secrets to logs
 		"environment", envs,
 		"working_directory", cmd.Dir,
+		"timeout", timeout,
 	)
 	cmd.Stderr = w
 	cmd.Stdout = w
-
+	// handling the timeout
+	if timeout > 0 {
+		// sets the same PGID for the child processes
+		setPGID(cmd)
+		terminationTimer := e.stopProcessWithTimeout(cmd, timeout)
+		err := cmd.Run()
+		// stop timer if command had finished before the timeout reached
+		terminationTimer.Stop()
+		return err
+	}
 	return cmd.Run()
```

Comment:
same as for killingTimer, and you can keep the exit point the same.


```suggestion
	if timeout > 0 {
		// sets the same PGID for the child processes
		setPGID(cmd)
		terminationTimer := e.stopProcessWithTimeout(cmd, timeout)
		defer func(t *time.Timer){
		    // stop timer if command had finished before the timeout reached
		    if !t.Stop() {
		        <-t.C
		    }
		}(terminationTimer)
	}
	return cmd.Run()
```